gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -269,32 +296,55 @@ private boolean isPathWhiteListed(String path)
                 || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
     }
 
-    public boolean handleAuthRequest(HttpServletRequest request)
+    @Override
+    public Optional<Cookie> removeCookie(HttpServletRequest clientRequest)
     {
-        return true;
+        if (!removeCookiePaths.contains(clientRequest.getRequestURI())) {
+            return Optional.empty();
+        }
+        Optional<Cookie> cookie = Arrays.stream(clientRequest.getCookies()).filter(
+                c -> c.getName().equalsIgnoreCase(COOKIE_NAME)).findAny();
+        clientRequest.getSession().invalidate();
+        cookie.ifPresent(c -> routingManager.removeCookie(c.getValue().split("\\.")[0]));
+
+        return cookie.map(c -> {
+            c.setMaxAge(0);
+            c.setValue("delete");
+            return c;
+        });
     }
 
     @Override
-    public String rewriteTarget(HttpServletRequest request)
+    public String rewriteTarget(HttpServletRequest request, int requestId)
     {
         if (!isPathWhiteListed(request.getRequestURI())) {
             return buildUriWithNewBackend(applicationEndpoint, request);
         }
 
-        if (isUseStickyRouting(request)) {
-            String backend = getPreviousBackend(request);
-            logRewrite(backend, request);
-            return buildUriWithNewBackend(backend, request);
+        Optional<String> previousBackend = getPreviousBackend(request);
+        if (previousBackend.isPresent()) {
+            previousBackend.ifPresent(b -> logRewrite(b, request));
+            return previousBackend.map(b -> buildUriWithNewBackend(b, request)).orElseThrow();
         }
 
         String backend = getBackendFromRoutingGroup(request);
-        // set target backend so that we could save queryId to backend mapping later.
         ((MultiReadHttpServletRequest) request).addHeader(PROXY_TARGET_HEADER, backend);
         logRewrite(backend, request);
 
         return buildUriWithNewBackend(backend, request);
     }
 
+    private void setCookie(HttpServletRequest request, HttpServletResponse response, String backend)
+    {
+        if (!cookiePaths.contains(request.getRequestURI())) {
```

Comment:
Docs states this is a startsWith check, but the actual logic is doing a full match check?